src/Views/Chat.hs

```diff
@@ -0,0 +1,113 @@
+module Views.Chat
+    (
+    listenForChatMessages,
+    chatMessageCallback,
+    createChatDOM
+    ) where
+
+import Haste.App
+import Haste
+import Haste.Events
+import Haste.DOM
+import LobbyTypes
+import LobbyAPI
+import Control.Monad (unless)
+
+-- | Chat dom creation.
+createChatDOM :: LobbyAPI -> Elem -> Client ()
+createChatDOM api parentDiv = do
+
+  br <- newElem "br"
+
+  chatDiv <- newElem "div" `with`
+    [
+      attr "id" =: "chatDiv"
+    ]
+
+  chatBox <- newElem "textarea" `with`
+    [
+      attr "id"       =: "chatBox",
+      attr "rows"     =: "10",
+      attr "cols"     =: "18",
+      attr "readonly" =: ""
```

Comment:
Tthe `readonly` attribute doesn't require any value. if present, with or without value, it makes the TextArea readonly. Valueless attributes can't be added with `Haste.DOM` so i'll just set the value to True to make it clearer.